lib/model/LoginStatus.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:openfoodfacts/interface/JsonObject.dart';
+
+/// Status after an attempt to log in.
+///
+/// Typical JSON:
+/// * wrong user and/or password
+///   {
+///     "status":0,
+///     "status_verbose":"user not signed-in"
+///   }
+/// * successful login:
+///   {
+///     "status":1,
+///     "status_verbose":"user signed-in",
+///     "user_id":"gqwbgsvvod",
+///     "user":{
+///       "email":"blababla@gmail.com",
+///       "name":"Mr. John Doe"
+///     }
+///   }
+class LoginStatus {
+  LoginStatus({
+    required this.status,
+    required this.statusVerbose,
+    this.userEmail,
+    this.userName,
+    this.userId,
+  });
+
+  final int status;
+  final String statusVerbose;
+  final String? userEmail;
+  final String? userName;
+  final String? userId;
+
+  factory LoginStatus.fromJson(Map<String, dynamic> json) => LoginStatus(
+        status: JsonObject.parseInt(json['status'])!,
+        statusVerbose: json['status_verbose'] as String,
```

Comment:
We were talking about not failing if some parameter is missing, we could make this nullable but to be honest it's not that important here